canon.go

```diff
@@ -91,3 +91,38 @@ func canonValue(chain *chain, in interface{}) (interface{}, bool) {
 
 	return out, true
 }
+
+func canonDecode(chain *chain, value interface{}, target interface{}) {
+	if chain.failed() {
+		return
+	}
+
+	if target == nil {
+		chain.fail(AssertionFailure{
+			Type:   AssertNil,
+			Actual: &AssertionValue{target},
+			Errors: []error{
+				errors.New("expected: target should not be nil"),
+			},
+		})
+	}
+	b, err := json.Marshal(value)
+	if err != nil {
+		chain.fail(AssertionFailure{
+			Type:   AssertValid,
+			Actual: &AssertionValue{value},
+			Errors: []error{
+				errors.New("expected: marshable value"),
+			},
+		})
+	}
+	if err := json.Unmarshal(b, target); err != nil {
+		chain.fail(AssertionFailure{
+			Type:   AssertValid,
+			Actual: &AssertionValue{target},
+			Errors: []error{
+				errors.New("expected: unmarshable value"),
+			},
+		})
```

Comment:
Let's reword this: `expected: value can be unmarshaled into target argument`. Instead of AssertValid, let's use AssertType, it will include target type in error message.